src/genie/libs/parser/iosxr/show_controllers.py

```diff
@@ -1286,3 +1287,481 @@ def cli(self, unit=0, location='all', output=None):
+# Schema for:
+#   'show controllers { ethernet_interface_type } \
+#   {interface_path_id} phy'
+# Where:
+#   ethernet-interface-type: GigabitEthernet | HundredGigE | TenGigE
+#   interfaces-path-id:  R/S/I/P  Forward interface in Rack/Slot/Instance/Port format
+# =====================================================================================
+class ShowControllers_WORD__WORD_Phy(ShowControllers_WORD__WORD_Phy_Schema):
+    valid_ethernet_interface_types = ("gigabitethernet", "hundredgige", "tengige")
+    cli_command = ['show controllers {ethernet_interface_type} {interface_path_id} phy',
+        'show controllers {ethernet_interface_type} phy',
+        'show controllers phy']
+
+    def cli(self, ethernet_interface_type=None, interface_path_id=None, output=None):
+        if output is None:
+            if ethernet_interface_type and ethernet_interface_type.lower() in self.valid_ethernet_interface_types:
+                if interface_path_id:
+                    cmd = self.cli_command[0].format(ethernet_interface_type=ethernet_interface_type, 
+                    interface_path_id=interface_path_id)
+                    out = self.device.execute(cmd)
+                    return self.cli_parse(out, ethernet_interface_type, interface_path_id)
+                else:
+                    interfaces_list = self.cli_get_interfaces(ethernet_interface_type)
+            else:
+                interfaces_list = []
+                for ethernet_interface_type in self.valid_ethernet_interface_types:
+                    interfaces_list += self.cli_get_interfaces(ethernet_interface_type)
+            return_dict = {}
+            for ethernet_interface_type, interface_path_id in interfaces_list:
+                cmd = self.cli_command[0].format(ethernet_interface_type=ethernet_interface_type, 
+                    interface_path_id=interface_path_id)
+                out = self.device.execute(cmd, error_pattern=[])
+                return_dict.update(self.cli_parse(out, ethernet_interface_type, interface_path_id))
+            return return_dict
+        else:
+            out = output
+            if ethernet_interface_type and ethernet_interface_type.lower() in self.valid_ethernet_interface_types:
+                if interface_path_id:
+                    return self.cli_parse(out, ethernet_interface_type, interface_path_id)
+                else:
+                    interfaces_list = self.cli_get_interfaces(ethernet_interface_type)
+            else:
+                interfaces_list = []
+                for ethernet_interface_type in self.valid_ethernet_interface_types:
+                    interfaces_list += self.cli_get_interfaces(ethernet_interface_type)
+            return_dict = {}
+            for ethernet_interface_type, interface_path_id in interfaces_list:
+                return_dict.update(self.cli_parse(out, ethernet_interface_type, interface_path_id))
+            return return_dict
```

Comment:
Please see how other parsers structure this section. 

A common efficient way of doing this part is to build the string as you go and then only have one line with device.execute and no returns in the middle